### PR TITLE
UI/view state

### PR DESCRIPTION
**Pull request by NancyMcCallB**
When a view state is passed in to the ViewportComponent, clone it to make sure that no ViewStates are used in more than a single Viewport.

### CODE REVIEW

File: ui/components/src/ui-components/viewport/ViewportComponent.tsx [hunk 1]
```diff
@@ -163,13 +183,16 @@ export class ViewportComponent extends React.Component<ViewportProps, ViewportSt
...
+        return viewState;
       }
     } else {
       Logger.logError(UiComponents.loggerCategory(this), `Either viewDefinitionId or viewState must be provided as a ViewportComponent Prop`);
       return undefined;
     }
 
-    return viewState;
+    const clonedViewState = await ViewportComponent.cloneViewState(this.props.imodel, viewState);
+    return clonedViewState;
```
Comment (bsteinbk) [inline ui/components/src/ui-components/viewport/ViewportComponent.tsx:195]: nitpick - I believe you could have just done...
return ViewportComponent.cloneViewState(this.props.imodel, viewState);

**Review by bsteinbk — APPROVED** [review]

File: ui/components/src/ui-components/viewport/ViewportComponent.tsx [hunk 2]
```diff
@@ -150,6 +153,23 @@ export class ViewportComponent extends React.Component<ViewportProps, ViewportSt
     }
   }
 
+  private static async cloneViewState(imodelConnection: IModelConnection, viewState: ViewState): Promise<ViewState | undefined> {
+    const className = viewState.classFullName;
```
Comment (pmconne) [inline ui/components/src/ui-components/viewport/ViewportComponent.tsx:157]: Why not `return viewState.clone();`?

File: ui/components/src/ui-components/viewport/ViewportComponent.tsx (hunk 1 above)
Comment (NancyMcCallB) [reply ui/components/src/ui-components/viewport/ViewportComponent.tsx:195]: Correct, I could have. I added the extra line when I was debugging -- I'll change it.

File: ui/components/src/ui-components/viewport/ViewportComponent.tsx (hunk 2 above)
Comment (NancyMcCallB) [reply ui/components/src/ui-components/viewport/ViewportComponent.tsx:157]: I didn't find viewState.clone()! Ugh.

Comment (pmconne) [reply ui/components/src/ui-components/viewport/ViewportComponent.tsx:157]: Yeah it's way down on `EntityState`. It's not `async` either so you can probably just delete your `cloneViewState` entirely.

Comment (NancyMcCallB) [reply ui/components/src/ui-components/viewport/ViewportComponent.tsx]: I did exactly that -- removed my cloneViewState() and returned viewState.clone().

**Pull request merged**